### DISCUSSION TITLE
problem with Builder

### DISCUSSION THREAD

**Discussion by R3B4T0** [lcobucci/jwt#757] [Q&A] [answered]
How I can create the params ClaimsFormatter and Encoder to pass them to Builder constructor??

Error: "Too few arguments to function Lcobucci\JWT\Token\Builder::__construct(), 0 passed in C:\xampp\htdocs\symfony\proyectoBassClefSymfony\src\Controller\ChatController.php on line 27 and exactly 2 expected"

Code: 

```php
public function chat()
    {
        $key = InMemory::plainText('koshensx');
        $username = $this->getUser()->getUsername();
        $token = (new Builder())
            ->withClaim('mercure', ['subscribe' => [sprintf("/%s", $username)]])
            ->getToken(
                new Sha256(),
                new Key($key)
            )
        ;
```

**Comment by Ocramius** [comment, marked as answer]
By default, the `Builder` is something you can get from a `Lcobucci\JWT\Configuration` instance, as explained in https://github.com/lcobucci/jwt/blob/207955dfbce28b4fbdc12367fbccf9546c66eb2d/docs/issuing-tokens.md

The `Configuration#builder()` uses a closure internally to give you a clean `Builder` each time:

https://github.com/lcobucci/jwt/blob/207955dfbce28b4fbdc12367fbccf9546c66eb2d/src/Configuration.php#L104-L107

https://github.com/lcobucci/jwt/blob/207955dfbce28b4fbdc12367fbccf9546c66eb2d/src/Configuration.php#L47-L49

**Reply by R3B4T0** [reply]
Thank you! But, What is the $container?? Because I get the error that the variable is not initialized

**Reply by Slamdunk** [reply]
Try following step-by-step the guide at https://lcobucci-jwt.readthedocs.io/en/stable/
